src/mme-app/utils/mmeNasUtils.cpp

```diff
@@ -2002,7 +2321,24 @@ void MmeNasUtils::encode_nas_msg(struct Buffer *nasBuffer, struct nasPDU *nas, S
 							nasBuffer->pos - mac_data_pos,
 							&nasBuffer->buf[mac_data_pos - MAC_SIZE], int_alg);
 			break;
-		} 
+		}
+		case ActivateDedicatedBearerContextRequest:
+		{
+			log_msg(LOG_DEBUG, "Encoding Activate Dedicated Bearer Context Req NAS message in mme-app\n");
+			uint8_t mac_data_pos = MmeNasUtils::encode_act_ded_br_req(nasBuffer, nas);
+
+			/* Calculate mac */
+                        uint8_t direction = 1;
+                        uint8_t bearer = 0;
+                        unsigned char int_key[NAS_INT_KEY_SIZE];
+                        secContext.getIntKey(&int_key[0]);
+            		nas_int_algo_enum int_alg = secContext.getSelectIntAlg();
```

Comment:
Is this selecting new algorithm or just reading selecting new algorithm. For sure we can not change algorithm while creating dedicated bearers.

Reply:
We are just reading the selected algorithm. We aren't changing it here.